### DISCUSSION TITLE
Is there a way to exclude volumes by name globally in the Restic opt-out strategy?

### DISCUSSION THREAD

**Discussion by doughepi** [vmware-tanzu/velero#3323] [Community support Q&A] [answered]
We're using Istio on the cluster, so backups get pretty messy with all of the Istio sidecar volumes.

It also seems to cause some failures. I'm not sure if the failure is from the nature of the Istio volumes or not. 

```
Failed:
    argocd/argocd-server-76cd89cf89-c6nwf: static-files
    argocd/argocd-server-76cd89cf89-s9x44: static-files
    jenkins-azuredevops-sync-env-prod/jenkins-azuredevops-sync-1611355200-ftnsk: istio-data, istio-envoy, istio-podinfo
    jenkins-azuredevops-sync-env-prod/jenkins-azuredevops-sync-1611356400-8hn6n: istio-data, istio-envoy, istio-podinfo
    jenkins-azuredevops-sync-env-prod/jenkins-azuredevops-sync-1611357600-j8jbl: istio-data, istio-envoy, istio-podinfo
    jenkins-azuredevops-sync-env-prod/jenkins-azuredevops-sync-1611358800-8hccn: istio-data, istio-envoy, istio-podinfo
```

There are plenty of successful backups of the Istio volumes. I'm mainly interested in excluding them as they are superfluous.

I'm totally fine with the answer being "add the exclude volume annotation to every Pod". It's just that, in practice, we don't have control over how the users annotate their pods. We would need to look into a mutating webhook tool like KubeMod.

Just wondering if this is possible using native Velero :).

**Comment by dsu-igeek** [comment, marked as answer]
At the moment, no.

**Comment by rdagerfall** [comment]
Anything new on this topic? I'm struggling with the same problem and would really like this feature to be implemented.

**Comment by qiuming-best** [comment]
maybe you could try [this](https://velero.io/docs/v1.13/resource-filtering/#resource-policies) one

**Comment by sseago** [comment]
@hazzel23 At the moment, we don't have a fix for this. The current velero release includes volume resource policies which allow users to control what happens to volumes at backup time based on certain criteria. There are two obstacles to using it for your purposes, though (one fixed in 1.14, the other would be a future enhancement):
1) The only action supported in Velero 1.13 is "skip"  -- different from opt-out, this means "don't back up the content at all".
2) Selection criteria includes volume capacity, storage class, and volume sources. We don't (yet) provide the ability to filter on name.

Of the two issues above, 1) will be resolved in 1.14, as we're providing the ability to do the equivalent of opt-out (i.e. use snapshots instead) using volume policies.
The one remaining thing needed to do what you want here would be to extend the selection criteria for volume policies to include PVC name (preferably with wildcard support). This would need to be a future enhancement for Velero 1.15 or later.
